repository.go

```diff
@@ -77,35 +77,70 @@ type Repository struct {
 	wt billy.Filesystem
 }
 
-type InitOptions struct {
-	// The default branch (e.g. "refs/heads/master")
-	DefaultBranch plumbing.ReferenceName
+type initOptions struct {
+	defaultBranch plumbing.ReferenceName
+	workTree      billy.Filesystem
+	objectFormat  formatcfg.ObjectFormat
+}
+
+func newInitOptions() initOptions {
+	return initOptions{
+		defaultBranch: plumbing.Master,
+		workTree:      nil,
+		objectFormat:  "",
+	}
+}
+
+type InitOption func(*initOptions)
+
+// WithDefaultBranch sets the default branch for the new repo (e.g. "refs/heads/master").
+func WithDefaultBranch(b plumbing.ReferenceName) InitOption {
+	return func(o *initOptions) {
+		o.defaultBranch = b
+	}
+}
+
+// WithWorkTree sets the internal worktree for the new repo.
+func WithWorkTree(worktree billy.Filesystem) InitOption {
+	return func(o *initOptions) {
+		o.workTree = worktree
+	}
+}
```

Comment:
```suggestion
// WithWorkTree sets the worktree filesystem for the repo. If not used, or a `nil` is
// passed as argument, will result in a bare repository.
func WithWorkTree(worktree billy.Filesystem) InitOption {
	return func(o *initOptions) {
		o.workTree = worktree
	}
}
```

Reply:
will remote `WithBare(..)` and incorporate this comment update in my rebased PR.

given the updated comments on `WithWorkTree(..)` what is the significance of keeping the `isBare` argument in
```
func PlainInit(path string, isBare bool, options ...InitOption) (*Repository, error) {
   ...
}
```

perhaps it makes sense to remove `isBare bool` there and leave the signature as
```
func PlainInit(path string, options ...InitOption) (*Repository, error) {
   ...
}
```
so that there is one consistent syntax for a repository as both `Init` and `PlainInit` default to a bare repo unless a worktree is passed using `WithWorkTree(...)`

Reply:
also, tests are failing for me locally on the `v6-transport` branch before I rebase my changes...

```
❯ make test
running against git version 2.35.1
...
--- FAIL: TestProxyEnvSuite (0.35s)
    --- FAIL: TestProxyEnvSuite/TestCommand (0.34s)
        testing.go:1399: race detected during execution of test
FAIL
FAIL	github.com/go-git/go-git/v6/internal/transport/ssh/test	2.317s
...
```